typographer/backends/skia/text_frame_skia.cc

```diff
@@ -29,14 +29,17 @@ static Font ToFont(const SkFont& font, Scalar scale) {
   return Font{std::move(typeface), std::move(metrics)};
 }
 
-TextFrame TextFrameFromTextBlob(sk_sp<SkTextBlob> blob,
-                                Scalar scale) {
+TextFrame TextFrameFromTextBlob(sk_sp<SkTextBlob> blob, Scalar scale) {
   if (!blob) {
     return {};
   }
 
   TextFrame frame;
 
+  auto bounds = blob->bounds();
```

Comment:
I don't think these bounds are particularly effective. There were instances where these were way off along at least one axis. I ran into this when adding placeholders for text (just solid boxes) as I was implementing glyph rendering.

I think a better approach would be to calculate these from the text frame using individual glyph positions and the conservative glyph sizes (we use these in the atlases as well). This way, we won't need to depend on backends to provide us an accurate bounds and our bounds will be tighter as well.

Reply:
Yeah that's what I did before I checked to see if it was already precomputed somewhere at parse time. My one test case is accurate, but it's a shame those bounds are unreliable. :/ Will replace it when I get a moment today.

Reply:
Done.

Reply:
Also made it optional/return nullopt when there are no runs and/or glyph positions in the frame.